packages/firebase_storage/ios/Classes/FirebaseStoragePlugin.m

```diff
@@ -92,6 +109,33 @@ - (void)getData:(FlutterMethodCall *)call result:(FlutterResult)result {
             }];
 }
 
+- (void)getMetadata:(FlutterMethodCall *)call result:(FlutterResult)result {
+  NSString *path = call.arguments[@"path"];
+  FIRStorageReference *ref = [[FIRStorage storage].reference child:path];
+  [ref metadataWithCompletion:^(FIRStorageMetadata *metadata, NSError *error) {
+    if (error != nil) {
+      result(error.flutterError);
+    } else {
+      NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] init];
+      [dictionary setValue:[metadata bucket] forKey:@"bucket"];
+      [dictionary setValue:[NSString stringWithFormat:@"%lld", [metadata generation]]  forKey:@"generation"];
+      [dictionary setValue:[NSString stringWithFormat:@"%lld", [metadata metageneration]] forKey:@"metadataGeneration"];
+      [dictionary setValue:[metadata path] forKey:@"path"];
+      [dictionary setValue:@([[metadata timeCreated] timeIntervalSince1970]) forKey:@"creationTimeMillis"];
```

Comment:
Isn't `timeIntervalSince1970` a `double` and measures seconds? -- In contrast to the Java side where the corresponding piece of information is a `long` and measures milliseconds. I think we should convert to a `long` by multiplying by 1000.0 and truncating, because Java and Dart agree on representing time using milliseconds and an integral datatype.